ui-speedspacechart/src/components/helpers/drawElements/tickY.ts

```diff
@@ -4,44 +4,49 @@ import { clearCanvas, speedRangeValues } from '../../utils';
 
 const { MARGIN_LEFT, MARGIN_TOP, MARGIN_BOTTOM, CURVE_MARGIN_TOP } = MARGINS;
 
+/** Provide the tick scale of the Y axis given the max speed */
+export const tickScaleY = (maxSpeed: number) => {
+  let tickScale = 5;
+  if (maxSpeed > 250) {
+    tickScale = 30;
+  } else if (maxSpeed > 130) {
+    tickScale = 20;
+  } else if (maxSpeed > 60) {
+    tickScale = 10;
+  }
+  return tickScale;
+};
+
 export const drawTickY = ({ ctx, width, height, store }: DrawFunctionParams) => {
-  const { speeds } = store;
+  const { maxSpeed } = speedRangeValues(store);
 
   clearCanvas(ctx, width, height);
 
-  const { maxSpeed } = speedRangeValues(store);
-
   ctx.strokeStyle = 'rgb(121, 118, 113)';
   ctx.lineWidth = 0.5;
   ctx.font = 'normal 12px IBM Plex Sans';
   ctx.fillStyle = 'rgb(182, 179, 175)';
   const textOffsetX = 36;
-  const textOffsetY = 24;
-
-  // horizontal ticks based on 10 units of round max speed
 
-  const yPosition = (height - MARGIN_BOTTOM - MARGIN_TOP - CURVE_MARGIN_TOP) / maxSpeed;
+  const tickScale = tickScaleY(maxSpeed);
+  const nbTicks = Math.ceil(maxSpeed / tickScale);
+  const maxTickSpeed = nbTicks * tickScale;
+  const ratioRoundPositions = maxTickSpeed / maxSpeed;
+  const ticksOffset =
+    ((height - MARGIN_BOTTOM - MARGIN_TOP - CURVE_MARGIN_TOP) * ratioRoundPositions) / nbTicks;
 
   ctx.beginPath();
-  speeds.forEach((_, i) => {
-    if (i >= 0 && i * 10 <= maxSpeed + 10) {
-      ctx.moveTo(43, height - MARGIN_BOTTOM - yPosition * i * 10);
-      ctx.lineTo(MARGIN_LEFT, height - MARGIN_BOTTOM - yPosition * i * 10);
-      ctx.textAlign = 'right';
-      const text = (i * 10).toString();
-      const textPositionY = height - MARGIN_TOP + 2 - textOffsetY - yPosition * i * 10;
-      let opacity = 1;
-
-      // low progressivily opacity for text when text is near borders, except for 0
-      if (textPositionY < MARGIN_TOP + 30)
-        opacity = (10 - (MARGIN_TOP + 30 - textPositionY) / 3) / 10;
-      if (textPositionY > height - MARGIN_BOTTOM - 30)
-        opacity = (10 - (height - MARGIN_BOTTOM - 30 - textPositionY) / 3) / 10;
-
-      ctx.fillStyle = `rgba(182, 179, 175, ${opacity})`;
-      ctx.fillText(text, textOffsetX, textPositionY);
-    }
-  });
+  for (let i = 0; i <= nbTicks; i++) {
+    const positionY = height - MARGIN_BOTTOM - ticksOffset * i;
+    ctx.moveTo(42, positionY);
```

Comment:
To avoid the magic number, maybe we can add a `const TICK_WIDTH = 6;` and then 

```suggestion
    ctx.moveTo(MARGIN_LEFT - TICK_WIDTH, positionY);
```